.travis.yml

```
@@ -274,6 +274,12 @@ matrix:
         - ! git grep -q --ignore-case "gnu general public";
         - ! git grep -q --ignore-case "gnu library general public";
 
+    - env:
+        - NAME=include_check
+      script:
+        - echo 'Checking that there are no '#include "mbed.h"' in code where it should not be'
+        - ! git grep '^#include "mbed.h"$' -- '*.c' '*.h' '*.cpp' '*.hpp' ':!*platform_mbed.h' ':!*TESTS/*' ':!TEST_APPS/' ':!UNITTESTS/' ':!*tests/*' ':!*targets/*' ':!*TARGET_*' ':!*unsupported/*'
```

@0xc0170 Could you confirm that astyle settings will guarantee that `^#include "mbed.h"$` is sufficient for matching? 

Otherwise, @alekla01 might need to make this search a bit more generic, maybe something like the following:

```suggestion
        - ! git grep '^#include.*mbed.h["']$' -- '*.c' '*.h' '*.cpp' '*.hpp' ':!*platform_mbed.h' ':!*TESTS/*' ':!TEST_APPS/' ':!UNITTESTS/' ':!*tests/*' ':!*targets/*' ':!*TARGET_*' ':!*unsupported/*'
```

---

> @0xc0170 Could you confirm that astyle settings will guarantee that ^#include "mbed.h"$ is sufficient for matching?

I don't think it formats this. Thus between include and mbed should be "any space"